src/nbl/video/COpenGLRenderpassIndependentPipeline.h

```diff
@@ -106,30 +106,11 @@ class COpenGLRenderpassIndependentPipeline final : public IGPURenderpassIndepend
 
 	        inline std::atomic_uint32_t& getStamp(IGPUSpecializedShader::E_SHADER_STAGE _stage)
 	        {
-		        switch (_stage)
-		        {
-		            case IGPUSpecializedShader::ESS_VERTEX:
-			            return *(stageUpdateStamps+0u);
-			            break;
-		            case IGPUSpecializedShader::ESS_TESSELATION_CONTROL:
-			            return *(stageUpdateStamps+1u);
-			            break;
-		            case IGPUSpecializedShader::ESS_TESSELATION_EVALUATION:
-			            return *(stageUpdateStamps+2u);
-			            break;
-		            case IGPUSpecializedShader::ESS_GEOMETRY:
-			            return *(stageUpdateStamps+3u);
-			            break;
-		            case IGPUSpecializedShader::ESS_FRAGMENT:
-			            return *(stageUpdateStamps+4u);
-			            break;
-		            default:
-			            break;
-		        }
-		        assert(false);
-		        return *(stageUpdateStamps+IGPURenderpassIndependentPipeline::SHADER_STAGE_COUNT);
+                return const_cast<std::atomic_uint32_t&>(
+                    const_cast<const PushConstantsState*>(this)->getStamp(_stage)
+                );
 	        }
-            inline const uint32_t& getStamp(IGPUSpecializedShader::E_SHADER_STAGE _stage) const
+            inline const std::atomic_uint32_t& getStamp(IGPUSpecializedShader::E_SHADER_STAGE _stage) const
```

Comment:
is the timestamp supposed to change value on us `volatile` style while we attempt to use it to compare with stuff?